docs/src/modules/components/overview/DateLibraries.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+// @ts-ignore
+import SectionHeadline from 'docs/src/components/typography/SectionHeadline';
+import { HighlightedCode } from '@mui/docs/HighlightedCode';
+import Button from '@mui/material/Button';
+import Divider from '@mui/material/Divider';
+import Stack from '@mui/material/Stack';
+import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
+import ToggleButton from '@mui/material/ToggleButton';
+import Typography from '@mui/material/Typography';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+
+const dateLibraries = [
+  {
+    name: 'Dayjs',
+    link: '/static/x/date-libraries/dayjs.png',
+    adapter: 'AdapterDayjs',
+    value: "dayjs('2024-04-17')",
+  },
+  {
+    name: 'Luxon',
+    link: '/static/x/date-libraries/luxon.png',
+    adapter: 'AdapterLuxon',
+    value: "DateTime.fromISO('2024-04-17')",
+  },
+  {
+    name: 'date-fns',
+    link: '/static/x/date-libraries/datefns.png',
+    adapter: 'AdapterDateFns',
+    value: "new Date('2024-04-17')",
+  },
+  {
+    name: 'Moment.js',
+    link: '/static/x/date-libraries/momentjs.png',
+    adapter: 'AdapterMoment',
+    value: "moment('2024-04-17')",
+  },
+];
+
+export default function DateLibraries() {
+  const [selectedLibrary, setSelectedLibrary] = React.useState(0);
+
+  const handleLibrarySwitch = (_event: React.MouseEvent<HTMLElement>, library: number) => {
+    if (library !== null) {
+      setSelectedLibrary(library);
+    }
+  };
+  return (
+    <React.Fragment>
+      <Divider />
+      <Stack direction={{ xs: 'column', md: 'row' }} spacing={6} py={4} alignItems="center">
+        <Stack spacing={2} sx={{ maxWidth: '450px' }}>
+          <SectionHeadline
+            overline="Date libraries"
+            title={
+              <Typography variant="h2" fontSize="1.625rem">
+                Support for multiple date libraries
```

Comment:
This might be more catchy, not sure which tone we want to take here
cc @samuelsycamore 

```suggestion
                Support your favorite date library
```